symja_android_library/matheclipse-core/src/main/java/org/matheclipse/core/reflection/system/ListLinePlot3D.java

```diff
@@ -29,140 +29,154 @@ public IExpr evaluate(final IAST ast, EvalEngine engine) {
-      } else if (dimension[1] > 3) {
-        final int valuesSize = values.size();
-        IASTAppendable resultList = F.ListAlloc(dimension[0]);
-        for (int i = 1; i < valuesSize; i++) {
-          IAST rowList = (IAST) values.get(i);
-          final int rowListSize = rowList.size();
-          IASTAppendable lineList = F.ListAlloc(dimension[1]);
-          final IAST color = GraphicsFunctions.plotStyleColorExpr(lineColorNumber[0]++, plotStyle);
-
-          for (int j = 1; j < rowListSize; j++) {
-            // ListLinePlot3D size is 2.5 × 2.5 × 1 independently from its coordinates
-            lineList.append(F.List(F.num(i * 2.5 / valuesSize), F.num(j * 2.5 / rowListSize),
-                rowList.get(j).divide(deltaHeight)));
-          }
-          resultList.append(color);
-          resultList.append(F.Line(lineList));
+      final IAST color = GraphicsFunctions.plotStyleColorExpr(lineColorNumber++, plotStyle);
+      resultList.append(color);
+      resultList.append(F.Line(lineList));
+    }
+    return resultList;
+  }
+
+  private IExpr coordinateLinePlot(IAST coordinates, IAST plotStyle, EvalEngine engine) {
+    double minX = ((IAST) (((IAST) coordinates.arg1()).arg1())).arg1().evalDouble();
+    double maxX = minX;
+
+    double minY = ((IAST) (((IAST) coordinates.arg1()).arg1())).arg2().evalDouble();
+    double maxY = minY;
+
+    double minZ = ((IAST) (((IAST) coordinates.arg1()).arg1())).arg3().evalDouble();
+    double maxZ = minZ;
+
+    for (int i = 1; i <= coordinates.argSize(); i++) {
+      IAST line = (IAST) coordinates.get(i);
+
+      for (int j = 1; j <= line.argSize(); j++) {
+        try {
+          IAST coordinate = (IAST) line.get(j);
+
+          // evalDouble may throw ArgumentTypeException
+          double x = coordinate.arg1().evalDouble();
+          if (x < minX)
+            minX = x;
+          if (x > maxX)
+            maxX = x;
+
+          double y = coordinate.arg2().evalDouble();
+          if (y < minY)
+            minY = y;
+          if (y > maxY)
+            maxY = y;
+
+          double z = coordinate.arg3().evalDouble();
+          if (z < minZ)
+            minZ = z;
+          if (z > maxZ)
+            maxZ = z;
+        } catch (ArgumentTypeException ate) {
+          // ignore this row
         }
-        return resultList;
       }
     }
-    return F.NIL;
+
+    // ListLinePlot3D size is 2.5 × 2.5 × 1 independently from its coordinates
+    IAST deltaXYZ =
+        F.List(F.num((maxX - minX) / 2.5), F.num((maxY - minY) / 2.5), F.num(maxZ - minZ));
+
+    // the color number with which the line will be printed
+    int lineColorNumber = 1;
+
+    IASTAppendable lineList = F.ListAlloc(coordinates.size() * 2);
+
+    for (int i = 1; i <= coordinates.argSize(); i++) {
+      final IAST color = GraphicsFunctions.plotStyleColorExpr(lineColorNumber++, plotStyle);
+
+      lineList.append(color);
+      // (# / deltaXYZ)& /@ line
+      lineList.append(F.Line(F.Map(
```

Comment:
The `S.Map.of(engine, ...)` syntax will evaluate immediately. Otherwise it will be evaluated in the rendering process of `Graphics3D`.
Maybe numeric evaluation with the `S.N.of(...)` function will also improve performance?


```java 
      lineList.append(
          F.Line(S.Map.of(engine, F.Function(F.Divide(F.Slot1, deltaXYZ)), coordinates.get(i))));
```

Reply:
Done.